pkg/helper/podHelper.go

```diff
@@ -0,0 +1,297 @@
+		option,
+		scheme.ParameterCodec,
+	)
+	exec, err := remotecommand.NewSPDYExecutor(restConfig, "POST", req.URL())
+	if err != nil {
+		return "", "", errors.Errorf("Failed executing command on throwaway pod %s", podName)
+	}
+
+	buf := &bytes.Buffer{}
+	errBuf := &bytes.Buffer{}
+
+	err = exec.StreamWithContext(context.Background(), remotecommand.StreamOptions{
+		Stdout: buf,
+		Stderr: errBuf,
+	})
+	if err != nil {
+		return "", "", errors.Errorf("Failed executing command on throwaway pod %s", podName)
+	}
+
+	return buf.String(), errBuf.String(), nil
+}
+
+func getClient() (*kube.Clientset, *restclient.Config, error) {
+
+	kubeCfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+	restCfg, err := kubeCfg.ClientConfig()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	kubeClient, err := kube.NewForConfig(restCfg)
+	if err != nil {
+		return nil, nil, errors.Errorf("Failed to generate new client for throwaway pod %s", err)
+	}
+	return kubeClient, restCfg, nil
+}
+
+func CreateRedisThrowAwayPod(k8sclient client.Client, namespace string) (*v1.Pod, error) {
+	// Create throwaway redis deployment
+	systemRedisPod := throwAwayRedis(namespace)
+
+	err := k8sclient.Create(context.TODO(), systemRedisPod)
+	if err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
+			// if it's other error than "AlreadyExists"
+			return systemRedisPod, err
+		}
+	}
+
+	// Wait for deployment to become ready
+	err = wait.Poll(time.Second*5, time.Minute*3, func() (done bool, err error) {
+		err = k8sclient.Get(context.TODO(), client.ObjectKey{Name: systemRedisPod.Name, Namespace: systemRedisPod.Namespace}, systemRedisPod)
+		if err != nil {
+			// Failed getting the deployment, trying again
+			return false, nil
+		}
+
+		for _, condition := range systemRedisPod.Status.Conditions {
+			if condition.Type == v1.PodReady {
+				if condition.Status == v1.ConditionTrue {
+					return true, nil
+				} else {
+					return false, nil
+				}
+			}
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		// Failed to boot up Redis throwaway pod
+		return systemRedisPod, err
+	}
+
+	return systemRedisPod, nil
+}
+
+func DeletePod(k8sclient client.Client, pod *v1.Pod) error {
+	err := k8sclient.Delete(context.TODO(), pod)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func CreateDatabaseThrowAwayPod(k8sclient client.Client, namespace, dbType string) (*v1.Pod, error) {
+	// Create throwaway redis deployment
+	dbPod := &v1.Pod{}
+	if dbType == "mysql" {
+		dbPod = throwAwayMysql(namespace)
+	}
+	if dbType == "postgres" {
+		dbPod = throwAwayPostgres(namespace)
+	}
+
+	err := k8sclient.Create(context.TODO(), dbPod)
+	if err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
+			// if it's other error than "AlreadyExists"
+			return dbPod, err
+		}
+	}
+
+	// Wait for deployment to become ready
+	err = wait.Poll(time.Second*5, time.Minute*3, func() (done bool, err error) {
```

Comment:
See above comment about replacing deprecated `wait.Poll` with `wait.PollUntilContextTimeout`

Reply:
If it's not a big deal, I'd rather stick to the old version for now and refactor when required.